engine-tests/src/tests/sanity.rs

```diff
@@ -135,6 +135,47 @@ fn test_log_address() {
     assert_eq!(log_address, greet_contract.address);
 }
 
+#[test]
+fn test_solidity_pure_bench() {
+    let (mut runner, mut signer, _) = initialize_transfer();
+    runner.wasm_config.limit_config.max_gas_burnt = u64::MAX;
+
+    let constructor = test_utils::solidity::ContractConstructor::force_compile(
+        "src/tests/res",
+        "target/solidity_build",
+        "bench.sol",
+        "Bencher",
+    );
+
+    let nonce = signer.use_nonce();
+    let contract = runner.deploy_contract(
+        &signer.secret_key,
+        |c| c.deploy_without_constructor(nonce.into()),
+        constructor,
+    );
+
+    // Number of iterations to do
+    let loop_limit = 100_000;
```

Comment:
I know you said you wanted 10^6 iterations @matklad , but you can already see how ridiculous the costs are at just 10^5 😅